### PR TITLE
refactor/logging

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- service logs to console by default in production

# solution

- disable logging by default in production
- document how to configure logging
- document how newcomers can more easily run the tests

# discussion

thx for the great lib.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,18 +69,21 @@ pubsubInstance.once('channelName', function (channelPayload) {
...
-* **close** – closes down the database connection and removes all listeners. Useful for graceful shutdowns.
-* All [EventEmitter methods](http://nodejs.org/api/events.html#events_class_events_eventemitter) are inherited from `EventEmitter`
-
 ## Description
 
 Creating a `PGPubsub` instance will not do much up front. It will prepare itself to start a Postgres connection once the first channel is added and then it will keep a connection open until its shut down, reconnecting it if it gets lost, so that it can constantly listen for new notifications.
 
 ## Lint / Test
 
-`npm test` or to watch, install `grunt-cli` then do `grunt watch`
```
Comment (cdaringe) [inline README.md]: looks like grunt is no more in this project

File: README.md [hunk 2]
```diff
@@ -20,7 +19,30 @@ The `pg` npm package has to be installed. Also – at least node 0.12 or iojs ha
 
 ## Usage
 
-Simple:
```
Comment (cdaringe) [inline README.md]: i hope these changes weren't offensive.  i...

- added docs for import and construction,
- added docs for the `options`,
- grouped all the API related things next to each other, then grouped examples together below.

**Comment by voxpelli** [comment]
Thanks 🙏

Looks like overall good changes. I’ll try to give some feedback as soon as possible, but can’t make a promise on any specific time frame.

If any particular changes are extra urgent, then please do remind/nag me about it until I do find some time, to ensure I don’t forget this 😉

**Comment by cdaringe** [comment]
not urgent at all.  in fact, a lot of this is opinionated, so i won't be upset if you don't love all the changes.  feel free to modify as seen fit it you like any of it.

**Comment by voxpelli** [comment]
Thanks for this @cdaringe and so sorry for dropping the ball on this. I've had so little time for opensource work during the last year, trying to wade through it all now. A release is coming up very soon!

**Comment by voxpelli** [comment]
@cdaringe Would love to re-converge – are there any other patches there that's missing here? Maybe you would be interested to co-maintain this module with me?

**Comment by cdaringe** [comment]
ya i did a few extra things as well:

1) made all of the tests setup their own database containers and tear them down on complete. further, i tightly coupled the tests to docker, and orchestrate it all programatically.
2) promisified the instance methods, e.g. made all the the subscribe/unsubscribe methods `async`.  for example, [uses like this](https://github.com/voxpelli/node-pg-pubsub/blob/master/index.js#L107-L120) now only support a promise by default, and, one can `await` until the query really does finish.  in the referenced lines of code, it's a race between user code and the `unlisten` query.  as part of ^ decision, i decided to only support newer versions of node, where async/await is ready out of the box.  9.x+ i think?
3) published typings, which are great for self documentation
4) wired in semantic-release

there were a few other smallies _i think_ (iirc around the retry logic), but i don't recall of hand!  your coverage is great--i [reused all of your tests](https://github.com/cdaringe/pg-subscribe/tree/master/src/test) and got ~full coverage off the bat!

i'd love to pair up!  i have recently become a TS fan and already have some deps on the TS version.  is that something you'd entertain?

**Comment by voxpelli** [comment]
@cdaringe Sorry for late response, so much going on right now.

I'm into TS-linting, but I'm mainly approaching it through JSDoc – see this project for an example of that: https://github.com/voxpelli/node-fetch-politely So I would definitely be into both using the type definitions of others and expose good type definitions to consumers.

1. Kind of like it, but would probably prefer it to be optional as I'm using a local postgres myself and because one can then continue to use the travis tests as they are today
2. It's okay from node 8+ with async/await – I would be totally okay with pushing a node 8+ only version and move to a fully promisified API
3. Typings are great 👌
4. I've worked with semantic releases before, so we could definitely do that

I'll add you here and on npm

**Comment by voxpelli** [comment]
You're added now @cdaringe 🙂

If you want to do any bigger changes here and now, can you open an issue or PR first and wait a few days for me to be able to give feedback and only if I haven't given feedback in a few days you go ahead and do it anyway?

**Pull request merged**